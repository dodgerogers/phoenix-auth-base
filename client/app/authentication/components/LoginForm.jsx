import React from 'react'
import OAuthSignInButton from '../containers/OAuthSignInButton';
import { Form, Header, Divider, Button, Icon } from 'semantic-ui-react';


function LoginForm(props) {
  return (
    <div className="login-form">
      <OAuthSignInButton
        fluid={true}
        color="facebook"
        provider="facebook"
      >
        <Icon name='facebook' /> Facebook
      </OAuthSignInButton>
      <div style={{ height: '5px' }} />
      <OAuthSignInButton
        fluid={true}
        color="google plus"
        provider="google_oauth2"
      >
        <Icon name='google' /> Google
      </OAuthSignInButton>
      <Divider horizontal>Or</Divider>
      <Form size='large'>
        <Form.Input
          fluid
          icon='mail'
          iconPosition='left'
          placeholder='Email'
        />
        <Form.Input
          fluid
          icon='lock'
          type='password'
          iconPosition='left'
          placeholder='Password'
        />
        <Button fluid size='large' color="teal">Login</Button>
      </Form>
    </div>
  );
}

export default LoginForm;
