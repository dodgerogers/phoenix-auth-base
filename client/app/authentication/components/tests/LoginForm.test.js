import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { PureComponent } from '../LoginForm';


describe('LoginForm', () => {
  const fullProps = {};

  const pureComponent = (props) => shallow(
    <PureComponent {...props} />
  );

  it('matches component snapshot', () => {
    const wrapper = pureComponent(fullProps);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
