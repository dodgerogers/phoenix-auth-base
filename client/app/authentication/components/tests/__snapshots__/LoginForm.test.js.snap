// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm matches component snapshot 1`] = `
<div
  className="login-form"
>
  <Form
    as="form"
    size="large"
  >
    <Field
      autoFocus={true}
      component={[Function]}
      fluid={true}
      icon="mail"
      iconPosition="left"
      name="email"
      placeholder="Email"
      validate={
        Array [
          [Function],
          [Function],
        ]
      }
    />
    <Field
      component={[Function]}
      fluid={true}
      icon="lock"
      iconPosition="left"
      name="password"
      placeholder="Password"
      type="password"
      validate={
        Array [
          [Function],
        ]
      }
    />
    <Button
      as="button"
      color="teal"
      disabled={true}
      fluid={true}
      type="submit"
    >
      Login
    </Button>
  </Form>
</div>
`;
